src/app/add-course-page/add-course-page.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { HomePageService } from '../core/services/home-page.service';
+import { CourseItemInfo } from '../core/models';
+
+@Component({
+  selector: 'app-add-course-page',
+  templateUrl: './add-course-page.component.html',
+  styleUrls: ['./add-course-page.component.less']
+})
+export class AddCoursePageComponent {
+
+  public title: string;
+  public description: string;
+  public duration: number;
+  public datapicker: Date;
+  public authors: string;
+
+  constructor (
+    private router: Router,
+    private homePageService: HomePageService
+  ) { }
+
+  public cancelIsClicked(): void {
+    this.router.navigate(['/courses']);
+  }
+
+  public addNewCourse(): void {
+    const sentDataCourse: CourseItemInfo[] = [{
+      title: this.title,
+      createdAtDate: this.datapicker,
+      durationTime: this.duration,
+      description: this.description,
+      authors: this.authors
+    }];
+
+    this.homePageService.createCourse(sentDataCourse);
+    this.router.navigate(['/courses']);
```

Comment:
can you assign value ['/courses'] to a variable?

Reply:
fixed